test/functional/ex_cmds/menu_spec.lua

```diff
@@ -379,5 +379,253 @@ describe('menu_get', function()
+            hidden = 0
+          },
+          {
+            priority = 500,
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "",
+                silent = 0
+              }
+            },
+            name = "Test7",
+            hidden = 0
+          },
+          {
+            priority = 500,
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "",
+                silent = 0
+              }
+            },
+            name = "Test8",
+            hidden = 0
+          },
+          {
+            priority = 500,
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "\"\"",
+                silent = 0
+              }
+            },
+            name = "Test9",
+            hidden = 0
+          }
+        },
+        priority = 500,
+        name = "Test"
+      }
+    }
+
+    eq(m, expected)
+  end)
+
+  it('works with right-aligned text and spaces', function()
+    clear()
+    command('nnoremenu &Test<Tab>Y.Test<Tab>X\\ x inormal<Alt-j>')
+    command('nnoremenu &Test\\ 1.Test\\ 2 Wargl')
+    command('nnoremenu &Test4.Test<Tab>3 i space<Esc>')
+
+    local m = funcs.menu_get("");
+    local expected = {
+      {
+        shortcut = "T",
+        hidden = 0,
+        actext = "Y",
+        submenus = {
+          {
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "inormal<Alt-j>",
+                silent = 0
+              }
+            },
+            hidden = 0,
+            actext = "X x",
+            priority = 500,
+            name = "Test"
+          }
+        },
+        priority = 500,
+        name = "Test"
+      },
+      {
+        shortcut = "T",
+        hidden = 0,
+        submenus = {
+          {
+            priority = 500,
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "Wargl",
+                silent = 0
+              }
+            },
+            name = "Test 2",
+            hidden = 0
+          }
+        },
+        priority = 500,
+        name = "Test 1"
+      },
+      {
+        shortcut = "T",
+        hidden = 0,
+        submenus = {
+          {
+            mappings = {
+              n = {
+                sid = 1,
+                noremap = 1,
+                enabled = 1,
+                rhs = "i space<Esc>",
+                silent = 0
+              }
+            },
+            hidden = 0,
+            actext = "3",
+            priority = 500,
+            name = "Test"
+          }
+        },
+        priority = 500,
+        name = "Test4"
+      }
+    }
```

Comment:
By the way, wondering how do you create such lengthy dictionaries? For #7234 and previous PRs I used to write the whole thing by hand until I got too bored and finally added https://github.com/ZyX-I/neovim/blob/f2724f85ed32f20443cfaa3a636ec875c313b19f/test/helpers.lua#L310-L359 which may be used to dump lua tables and strings (no support for numbers yet because specifically for my PR I do not need them), but maybe there already is something existing.

Reply:
Steady hand and a needle...

Hrhr, no, I'm good, top of the file, first test, contains a comment that helps with this. Thanks for asking though :)